clustermesh: explicitly report zero remote nodes on connection release

tklauser opened the pull request:
rc.remoteNodes is set to nil a few lines above [1], thus calling
NumEntries will always return 0 and will thus set the
clustermesh_remote_cluster_nodes metric to zero. Make that explicit
instead of relying on ((*store.SharedStore)nil).NumEntries() to return
0.

[1] https://github.com/cilium/cilium/blob/ab21ecbd53546abbc8d472498f1bbfae22842ee5/pkg/clustermesh/remote_cluster.go#L120-L121

aanm commented on pkg/clustermesh/remote_cluster.go:
```diff
@@ -129,7 +129,7 @@ func (rc *remoteCluster) releaseOldConnection() {
 	backend := rc.backend
 	rc.backend = nil
 
-	rc.metricTotalNodes.WithLabelValues(rc.mesh.conf.Name, rc.mesh.conf.NodeName, rc.name).Set(float64(rc.remoteNodes.NumEntries()))
```
IF we are releasing the old connection, shouldn't we set it to zero? :thinking:

tklauser replied:
My (wrong) assumption was that we might want to report the last known value and we would not see any updates afterwards. But you're right, setting it to zero is correct when we release the old connection. Though it probably makes sense to explicitly set 0 instead of relying on `(*store.SharedStore)(nil).NumEntries()` to return 0.

tklauser committed:
clustermesh: explicitly report zero remote nodes on connection release

rc.remoteNodes is set to nil a few lines above [1], thus calling
NumEntries will always return 0 and will thus set the
clustermesh_remote_cluster_nodes metric to zero. Make that explicit
instead of relying on ((*store.SharedStore)nil).NumEntries() to return
0.

[1] https://github.com/cilium/cilium/blob/ab21ecbd53546abbc8d472498f1bbfae22842ee5/pkg/clustermesh/remote_cluster.go#L120-L121

Signed-off-by: Tobias Klauser <tobias@cilium.io>
